packages/core/src/amazonq/lsp/lspController.ts

```diff
@@ -0,0 +1,361 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as path from 'path'
+import * as fs from 'fs-extra'
+import * as crypto from 'crypto'
+import { getLogger } from '../../shared/logger/logger'
+import { CurrentWsFolders, collectFilesForIndex } from '../../shared/utilities/workspaceUtils'
+import fetch from 'node-fetch'
+import { LspClient } from './lspClient'
+import AdmZip from 'adm-zip'
+import { RelevantTextDocument } from '@amzn/codewhisperer-streaming'
+import { makeTemporaryToolkitFolder } from '../../shared/filesystemUtilities'
+import { CodeWhispererSettings } from '../../codewhisperer/util/codewhispererSettings'
+import { activate as activateLsp } from './lspClient'
+import { telemetry } from '../../shared/telemetry'
+import { isCloud9 } from '../../shared/extensionUtilities'
+import { globals } from '../../shared'
+import { AuthUtil } from '../../codewhisperer'
+import { isWeb } from '../../shared/extensionGlobals'
+
+function getProjectPaths() {
+    const workspaceFolders = vscode.workspace.workspaceFolders
+    if (!workspaceFolders || workspaceFolders.length === 0) {
+        throw Error('No workspace folders found')
+    }
+    return workspaceFolders.map(folder => folder.uri.fsPath)
+}
+
+export interface Chunk {
+    readonly filePath: string
+    readonly content: string
+    readonly context?: string
+    readonly relativePath?: string
+    readonly programmingLanguage?: string
+}
+
+export interface Content {
+    filename: string
+    url: string
+    hashes: string[]
+    bytes: number
+    serverVersion?: string
+}
+
+export interface Target {
+    platform: string
+    arch: string
+    contents: Content[]
+}
+
+export interface Manifest {
+    manifestSchemaVersion: string
+    artifactId: string
+    artifactDescription: string
+    isManifestDeprecated: boolean
+    versions: {
+        serverVersion: string
+        isDelisted: boolean
+        targets: Target[]
+    }[]
+}
+const manifestUrl = 'https://aws-toolkit-language-servers.amazonaws.com/temp/manifest.json'
```

Comment:
So the "temp" URL is not temporary? Can this tech debt be avoided, why not rename the URL?

Reply:
DEXP team told me to use this url.